xarray/test/test_dataset.py

```diff
@@ -2918,17 +2918,38 @@ def test_filter_by_attrs(self):
             self.assertEqual(new_ds[var].height, '10 m')
 
     def test_binary_op_join_setting(self):
-        dim = 'x'
-        align_type = "outer"
-        coords_l, coords_r = [0, 1, 2], [1, 2, 3]
-        missing_3 = xr.DataArray(coords_l, [(dim, coords_l)]).to_dataset(name='a')
-        missing_0 = xr.DataArray(coords_r, [(dim, coords_r)]).to_dataset(name='a')
-        with xr.set_options(arithmetic_join=align_type):
-            experimental = missing_0 + missing_3
-        missing_0_aligned, missing_3_aligned =\
-            xr.align(missing_0, missing_3, join=align_type)
-        control = missing_0_aligned + missing_3_aligned
-        self.assertDatasetEqual(experimental, control)
+        # arithmetic_join applies to data array coordinates
+        missing_2 = xr.Dataset({'x':[0, 1]})
+        missing_0 = xr.Dataset({'x':[1, 2]})
+        with xr.set_options(arithmetic_join='outer'):
+            actual = missing_2 + missing_0
+        expected = xr.Dataset({'x':[0, 1, 2]})
+        self.assertDatasetEqual(actual, expected)
+
+        # arithmetic join also applies to data_vars
+        ds1 = xr.Dataset({'foo': 1, 'bar': 2})
+        ds2 = xr.Dataset({'bar': 2, 'baz': 3})
+        expected = xr.Dataset({'bar': 4}) # default is inner joining
+        actual = ds1 + ds2
+        self.assertDatasetEqual(actual, expected)
+        
+        with xr.set_options(arithmetic_join='outer'):
+            expected = xr.Dataset({'foo':1, 'bar': 4, 'baz': 3})
```

Comment:
As noted above, should be `xr.Dataset({'foo': nan, 'bar': 4, 'baz': nan})` for consistent outer join semantics